app/models/manager_refresh/inventory/persister.rb

```diff
@@ -53,7 +53,9 @@ def initialize_inventory_collections
   #
   # @param options [Hash] Hash used for ManagerRefresh::InventoryCollection initialize
   def add_inventory_collection(options)
-    options[:parent] ||= manager
+    if !options.key?(:parent) && manager
+      options[:parent] ||= manager
```

Comment:
Checking for the presence of the key, makes it so you don't need the `||=` anymore, right?